src/mono/mono/metadata/icall-def.h

```diff
@@ -1107,7 +1107,7 @@ ICALL_TYPE(WAITH, "System.Threading.WaitHandle", WAITH_1)
 HANDLES(WAITH_1, "SignalAndWait_Internal", ves_icall_System_Threading_WaitHandle_SignalAndWait_Internal, gint32, 3, (gpointer, gpointer, gint32))
 HANDLES(WAITH_2, "Wait_internal", ves_icall_System_Threading_WaitHandle_Wait_internal, gint32, 4, (gpointer_ptr, gint32, MonoBoolean, gint32))
 
-#if defined(TARGET_WASM)
+#if !defined(ENABLE_NETCORE) && defined(TARGET_WASM)
```

Comment:
This won't change anything for netcore—that file is already legacy-specific, we have `icall-def.h` and `icall-def-netcore.h`. You want to put this guard around the function declaration in `icall-decl.h` and the implementation (in `icall.c`, I believe) instead.

Reply:
It is already guarded in those places

Reply:
I'd really like to get rid of the syncing so that we can clean up all this code it is pretty difficult to parse the defines while reading.

Reply:
Oh oops, didn't see that it's guarded so far out. I think this one should already be covered by https://github.com/mono/mono/blob/22e81d941c00d0106e358bcccae89cb634985174/mono/metadata/icall-def.h#L110-L112 as well, assuming https://github.com/mono/mono/blob/22e81d941c00d0106e358bcccae89cb634985174/mono/metadata/icall-def.h#L1137 is the end of that condition? I wish that was clearer though, we should really enforce marking what the `#endif`s correspond to in here and probably decouple those files entirely. I'll open an issue so I remember to take a look at cleaning that up.

Reply:
Didn't see your second comment there before the first reply, but I think the icall files in particular are hard to deal with due to the split and could be improved. I don't expect the sync to go away any time soon, so it's worth me putting in some effort to make this a bit less painful I think.

Reply:
cc: @lambdageek